malcolm/modules/ADCore/parts/positionlabellerpart.py

```diff
@@ -83,7 +83,7 @@ def on_configure(
         futures += child.put_attribute_values_async(
             dict(enableCallbacks=True, idStart=id_start)
         )
-        self.steps_up_to = completed_steps + steps_to_do
+        self.steps_up_to = generator.size
```

Comment:
I am not sure if this is correct when completed steps is not 0. Initially I changed it to
```
self.steps_up_to = completed_steps + generator.size
```
but then I got errors trying to access generator points with index > generator.size, so I changed it.

Reply:
Any thoughts on this @thomascobb ?

Reply:
```self.done_when_reaches = genertor.size - completed_steps```